common_types/src/schema.rs

```
@@ -1135,7 +1183,26 @@ impl Builder {
 
     /// Build the schema
     pub fn build(self) -> Result<Schema> {
-        let timestamp_index = self.timestamp_index.context(MissingTimestampKey)?;
+        let retry_timestamp_key = if self.timestamp_index.is_none() {
```

if timestamp_index is not set, we should return an Error just like before.